pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,27 @@
+require 'core/call_context'
+
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          CallContext.new(env).redirect(normalized_path(env['PATH_INFO']), 301)
```

Comment:
@bryanp calling `redirect` here, won't work because ('unless I'm doing something wrong') there are missing dependencies, and at that low level it seems that they are not yet loaded.
run this spec `rspec spec/unit/middleware/req_path_normalizer_spec.rb:43` so you have an idea.
I'm kind of stuck here, I tried to require the missing dependencies/files but things started to get messy so I stopped. So, please advice :)